src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -143,4 +143,34 @@ internal static void SetException(Activity activity, Exception ex, bool escaped
         activity.SetStatus(ActivityStatusCode.Error, statusDescription);
         activity.Dispose();
     }
+
+    internal static Activity? ImportStart(string copyFromCommand, NpgsqlConnector connector)
+    {
+        var dbName = connector.Settings.Database ?? "UNKNOWN";
+
+        var activity = Source.StartActivity(dbName, ActivityKind.Client);
```

Comment:
Would be nice if there was a mechanism to inject the span name, analogous to `CommandSpanNameProvider`.

Reply:
Discussed with @NinoFloris offline. While 'technically' one can filter via sampling/processors, it's either the API is extremely limited or you're still paying the allocations via processor's workaround. So I think it's better if we do the exact same thing as [otel does for HttpClient](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.Http/HttpClientTraceInstrumentationOptions.cs#L114).